### PR TITLE
Add python-installer package

### PR DESCRIPTION
**Pull request by lazka**
Untested, just made it build

### CODE REVIEW

File: mingw-w64-python-installer/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+}
+
+build() {
+  cd "${srcdir}/simple_launcher"
+  windres --input launcher.rc --output resources.res --output-format=coff
+
+  local _gui_cflags="$CFLAGS"
+  local _cli_cflags="$CFLAGS"
+  local _gui_lflags="-lshlwapi"
```
Comment (jeremyd2019) [inline mingw-w64-python-installer/PKGBUILD:37]: shouldn't there be an `-mwindows` in there somewhere? and should it have `-municode`?

Comment (lazka) [reply mingw-w64-python-installer/PKGBUILD:37]: ah, right, thanks

Comment (jeremyd2019) [reply mingw-w64-python-installer/PKGBUILD:37]: building the same launcher is done in mingw-w64-python-distlib I think

Comment (lazka) [reply mingw-w64-python-installer/PKGBUILD:37]: that's where I copied it from initially. the source is not the same (but the code might be)

**Comment by lazka** [comment]
The next steps are

* Bootstrap flit manually: https://flit.readthedocs.io/en/latest/bootstrap.html
* Port packages using pyproject2setuppy to build+installer
* Drop pyproject2setuppy

**Comment by jeremyd2019** [comment]
Flit is already packaged, why does it need to be bootstrapped again?

**Comment by lazka** [comment]
it uses the now abandoned pyproject2setuppy

**Comment by jeremyd2019** [comment]
It looks like this package has a mechanism that will work without patching to map arches to proper launchers.

One concern I had with these launchers vs the setuptools ones is that they apparently combine the launcher exe and the script, instead of putting a `-script.py` file next to the launcher, but still do shebang processing.  This will make trying to `sed` the shebang after the fact more fraught.  It might be necessary to patch the shebang-generating code instead.  I think the launcher supports finding the executable in the shebang on the path instead of an absolute path, don't remember if it had a mode to look in the same directory as the launcher though.  It seemed to have a bunch of different defines that could be tweaked for different functionality.

**Comment by jeremyd2019** [comment]
Most of that probably also applies to distlib, but I haven't noticed any package actually using the launchers from there yet.

**Comment by lazka** [comment]
> Update: it does support shebang relative to the launcher out-of-the-box with default defines. For that to work, shebang must start with literal string `<launcher_dir>\`

good find, then we maybe just have to patch this package

I found another issue which is blocking us using this: https://github.com/pradyunsg/installer/issues/98

**Comment by pradyunsg** [comment]
> Most of that probably also applies to distlib, but I haven't noticed any package actually using the launchers from there yet.

Really? pip does, whenever it builds via a wheel.

**Comment by lazka** [comment]
We don't use pip for packaging, we might though in the future.

**Comment by pradyunsg** [comment]
You’ll find this blog from a setuptools maintainer useful to read: https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html

**Comment by lazka** [comment]
Thanks, yes, we are trying. We have one package so far: https://github.com/msys2/MINGW-packages/blob/82ee778df82fc2dc2cb7a8411a7f84ae93407858/mingw-w64-python-typing_extensions/PKGBUILD#L39

Let's merge this and and add some hacks for the wrong prefix in the respective PKGBUILDs. After we dropped pyproject2setuppy we can create a packaging template using pip for the rest.

**Comment by lazka** [comment]
One downside of the new launcher is that stripping breaks it and we need to use "!strip" everywhere.

**Comment by jeremyd2019** [comment]
I guess that makes sense, as it just concatenates the script contents onto the exe.  If somebody were really bugged by this it would probably be possible to embed the script in a resource, I have done that in the past with a dynamic 'payload' on a .exe and it's pretty easy with Windows APIs, and then the exe format knows it's there so tools shouldn't discard it.

**Comment by lazka** [comment]
I guess adding resources would require some help from the toolchain? python-installer uses this just via the built launchers and pure Python..

**Comment by jeremyd2019** [comment]
https://docs.microsoft.com/en-us/windows/win32/menurc/adding-deleting-and-replacing-resources so ctypes or pywin32 (https://mhammond.github.io/pywin32/win32api__UpdateResource_meth.html) could do it.  but probably not worth it.

**Comment by jeremyd2019** [comment]
Just read the README for simple launcher, and it seems it uses python support for running a zip archive, so putting it in a resource probably wouldn't work.

**Comment by lazka** [comment]
Ah, right.

**Pull request merged**